force_wfmanager/left_side_pane/workflow_tree.py

```diff
@@ -283,16 +334,154 @@ def new_data_source(self, ui_info, object):
         if result is not None:
             object.add_data_source(result)
 
+    @verify
     def delete_data_source(self, ui_info, object):
         self.workflow_mv.remove_data_source(object.model)
 
+    @verify
     def edit_data_source(self, ui_info, object):
         # This is a live dialog, workaround for issue #58
         object.model.edit_traits(kind="livemodal")
 
+    @verify
     def new_layer(self, ui_info, object):
         self.workflow_mv.add_execution_layer(ExecutionLayer())
 
+    @verify
     def delete_layer(self, ui_info, object):
         """ Delete an element from the workflow """
         self.workflow_mv.remove_execution_layer(object.model)
+
+    @on_trait_change("selected_mv,verify_workflow_event")
+    def update_selected_error(self):
+        if self.selected_mv is None:
+            self.selected_error = 'No errors'
+        elif self.selected_mv.error_message == '':
+            self.selected_error = 'No errors'
+        else:
+            self.selected_error = self.selected_mv.error_message
+
+    @on_trait_change("workflow_mv.verify_workflow_event")
+    def received_verify_request(self):
+        """Checks if the root node of workflow tree is requesting a
+        verification of the workflow"""
+        self.verify_workflow_event = True
+
+    def _verify_workflow_event_fired(self):
```

Comment:
be wary of the _fired or _changed methods, they are known to bite because they bind at a different time. Stay consistent and always use on_trait_change.